### PR TITLE
libfabric: meta.platforms = lib.platforms.all and 1.13.0 -> 1.13.1

### PR DESCRIPTION
**Pull request by ShamrockLee**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bzizou — APPROVED** [review]
Ok for the upgrade and platfoms change. I initially put x86_64-linux in a conservative approach as I didn't know much about this lib

**Comment by ShamrockLee** [comment]
@bzizou Thank you a lot for reviewing this change.

I'm not familiar with it either, but I notice that Debian and RPM provides the aarch64 version of this package, and this library is also provided by Homebrew.

There is a dependency called `libpsm2` that only builds under `x86_64`. Debian seems to opt-in this dependency on the `x86-64` platform, and it doesn't appear inside the Homebrew formulae of libfabric.

More tests are required.

https://packages.debian.org/source/bullseye/libfabric
https://formulae.brew.sh/formula/libfabric#default

**Comment by ShamrockLee** [comment]
@ofborg build libfabric

**Comment by ShamrockLee** [comment]
@ofborg build libfabric mpich

**Comment by ShamrockLee** [comment]
@markuskowa @nbren12 Maybe you would like to take a look, as this is a dependency of `mpich`.

**Review by markuskowa — APPROVED** [review]
This should not be problematic for MPICH or OpenMPI.

File: pkgs/development/libraries/libfabric/default.nix [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
 
-  nativeBuildInputs = [ pkg-config autoreconfHook ] ;
+  nativeBuildInputs = [ pkg-config autoreconfHook ];
 
-  buildInputs = [ libpsm2 ] ;
+  buildInputs = lib.optional enablePsm2 libpsm2;
 
-  configureFlags = [ "--enable-psm2=${libpsm2}" ] ;
+  configureFlags =
+    [ (if enablePsm2 then "--enable-psm2=${libpsm2}" else "--disable-psm2") ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libfabric/default.nix:22]:
```suggestion
  configureFlags = [ (if enablePsm2 then "--enable-psm2=${libpsm2}" else "--disable-psm2") ];
```

**Comment by ShamrockLee** [comment]
Amended ste above suggessions into the commits.
The build result should not be affected.

**Comment by ShamrockLee** [comment]
@markuskowa Thank you.
This change should enable MPICH to build on `aarch64-linux`.

**Review by bzizou — APPROVED** [review]
Tested with OpenMPI and osu-micro-benchmarks on 2 nodes with Omnipath 100 intel interface
(with `--mca pml ^ucx` to force the use of psm2). Seems to work, but got errors with PMXI. I'm going to test an upgrade of libpmxi.

**Comment by markuskowa** [comment]
@ShamrockLee thanks! MPICH works now on `aarch64-linux`

**Pull request merged**